tests/integration/test_interaction.py

```diff
@@ -336,6 +344,86 @@ def test_keyboard_help(self):
         self.interact_with_keyboard_help()
 
 
+class EventsFiredTest(DefaultDataTestMixin, InteractionTestBase, BaseIntegrationTest):
+
+    def setUp(self):
+        mock = Mock()
+        context = patch.object(WorkbenchRuntime, 'publish', mock)
+        context.start()
+        self.addCleanup(context.stop)
+        self.publish = mock
+        super(EventsFiredTest, self).setUp()
+
+    def _get_scenario_xml(self):  # pylint: disable=no-self-use
+        return "<vertical_demo><drag-and-drop-v2/></vertical_demo>"
+
+    def test_loaded(self):
+        dummy, name, event_data = self.publish.call_args[0]
+        self.assertEqual(name, 'edx.drag_and_drop_v2.loaded')
+        self.assertEqual(
+                event_data, {
+                    'component_id': u'drag-and-drop-v2.drag-and-drop-v2.d0.u0',
+                    'user_id': 'student_1'
+                }
+        )
+
+    def test_picked_up(self):
+        self.parameterized_item_positive_feedback_on_good_move(self.items_map)
+        dummy, name, event_data = self.publish.call_args_list[1][0]
+        self.assertEqual(name, 'edx.drag_and_drop_v2.item.picked_up')
+        self.assertEqual(
+                event_data, {
+                    'component_id': u'drag-and-drop-v2.drag-and-drop-v2.d0.u0',
+                    'user_id': 'student_1',
+                    'item_id': 0,
+                }
+        )
+
+    def test_dropped(self):
+        self.parameterized_item_positive_feedback_on_good_move(self.items_map)
+        # Skipping to 3, since 2 is grade event.
```

Comment:
@Kelketek Maybe add a docstring to the class that lists the order of events (`loaded` + sequence of events that happen when you do a "good move")? You wouldn't need this comment anymore, and similar information for the remaining tests in this class would be available in one place.